### PR TITLE
EPMDJ-5891 Fixed Drill tracks junit4 failed tests as passed

### PR DESCRIPTION
**Pull request by Mazel-Tovr**
Need to add new TestResult status to test2code plugin

### CODE REVIEW

**Review by LikhanovMaxim — APPROVED** [review]

File: src/runtimeMain/kotlin/com/epam/drill/test/agent/TestListener.kt [hunk 1]
```diff
@@ -53,6 +59,17 @@ actual object TestListener {
         ThreadStorage.stopSession()
     }
 
+    private fun isNotFinalizeTestState(test: String?): Boolean =
```
Comment (ViktoriiaLebedeva) [inline src/runtimeMain/kotlin/com/epam/drill/test/agent/TestListener.kt:62]: Why did you extract this into a separate method?

File: src/runtimeMain/kotlin/com/epam/drill/test/agent/TestListener.kt [hunk 2]
```diff
@@ -11,12 +11,14 @@ import java.util.concurrent.ConcurrentHashMap
...
-    private val _ml = ConcurrentHashMap<String, MutableMap<String, Any>?>()
+    private val logger = Logging.logger(TestListener::class.java.simpleName)
+
+    private val testToTestInfo = ConcurrentHashMap<String, MutableMap<String, Any>?>()
 
     private fun addTestInfo(testId: String, vararg vals: Pair<String, Any>) {
         vals.forEach {
             val (paramName, value) = it
-            val meta = _ml[testId] ?: _ml.run {
+            val meta = testToTestInfo[testId] ?: testToTestInfo.run {
```
Comment (ViktoriiaLebedeva) [inline src/runtimeMain/kotlin/com/epam/drill/test/agent/TestListener.kt:21]: rename mmo as well

File: src/runtimeMain/kotlin/com/epam/drill/test/agent/TestListener.kt [hunk 3]
```diff
@@ -11,12 +11,14 @@ import java.util.concurrent.ConcurrentHashMap
 @Kni
 actual object TestListener {
 
-    private val _ml = ConcurrentHashMap<String, MutableMap<String, Any>?>()
+    private val logger = Logging.logger(TestListener::class.java.simpleName)
+
+    private val testToTestInfo = ConcurrentHashMap<String, MutableMap<String, Any>?>()
```
Comment (ViktoriiaLebedeva) [inline src/runtimeMain/kotlin/com/epam/drill/test/agent/TestListener.kt:16]: rename it to testsInfo

**Review by ViktoriiaLebedeva — CHANGES_REQUESTED** [review]

File: src/runtimeMain/kotlin/com/epam/drill/test/agent/TestListener.kt (hunk 1 above)
Comment (Mazel-Tovr) [reply src/runtimeMain/kotlin/com/epam/drill/test/agent/TestListener.kt:62]: for better readability 🤷‍♂

**Review by IgorKey — APPROVED** [review]

**Review by LikhanovMaxim — APPROVED** [review]

**Pull request merged**